### PR TITLE
Feat: Enable feature livewire emit event

### LINKED ISSUE #178: Notification emit on Event
I have this idea to improve on the notification, but i dont know typescript :p

Usually we use this to fire event and its awesome.
```php
public function save(): void
   {
      // ...
         $this->notification()->notify([
            // ...
           'onClose' => [
               'method' => 'firedEvent',
               'params' => 'onClose',
         ],

       ]);
    }
```

It would be great if we can emit from the Livewire.emit() event and listen from component.

```php
'onClose' => [
     'emit' => 'testMethod',  // just example to emit
     'params' => 'mydata',
],
```

on notifications event typescript ( i do know typescript, so.. :p )

`
Livewire.emit(options?.emit, ...params)
`

And listen from the component.

`protected $listeners = ['testMethod'];`

Thank you.

### PR DESCRIPTION
**Pull request by putera**
This will add the feature #178.

Example code :

```php
$this->notification()->notify([
        'title' => 'Title',
        'description' => 'Description is in here',
        'onClose' => [
            'emit' => 'fireEmit',
            'params' => 'on-close-emit-event',
        ]
]);
```

On the component :
```php
<?php

use Livewire\Component;

class TheComponent extends Component
{
    protected $listeners = ['fireEmit'];

    public function fireEmit($data)
    {
            // manipulate the data
           info($data);
    }
```

If you want to emit to another component, use this :

```php
$this->notification()->notify([
        'title' => 'Title',
        'description' => 'Description is in here',
        'onClose' => [
            'emit' => 'fireEmitToAnotherComponent',
            'to' => 'another-component',  // same as the directive @livewire('another-component')
            'params' => 'on-close-emit-event',
        ]
]);
```

on AnotherComponent class :

```php
<?php

use Livewire\Component;

class AnotherComponent extends Component
{
    protected $listeners = ['fireEmitToAnotherComponent'];

    public function fireEmitToAnotherComponent($data)
    {
            // manipulate the data
           info($data);
    }
```

Image Preview of the working Features
[The Preview](https://i.ibb.co/0h7HcGg/Untitled-Project.gif)

### CODE REVIEW

**Comment by putera** [comment]
@PH7-Jack why my test is failing. on my local machine its success. please help.

**Comment by ph7jack** [comment]
> @PH7-Jack why my test is failing. on my local machine its success. please help.

Sometimes is needed to re-run the tests, because the dusk breaks with livewire

**Comment by ph7jack** [comment]
@putera what's conflicts do this branch has? don't are showing for me

**Comment by putera** [comment]
> @putera what's conflicts do this branch has? don't are showing for me

Hi brother.
I think the conflict because i've modified the TypeScript and need to recompile the js to make sure the test script is working.

File modified :
- ts/notifications/actions.ts
- ts/notifications/events.ts
- ts/notifications/options.ts
- ts/notifications/parses.ts

i need to run `npm run build` to test the feature.

**Comment by ph7jack** [comment]
@putera Could you merge the main on your branch?

File: tests/Browser/Notifications/view.blade.php [hunk 1]
```diff
@@ -4,70 +4,45 @@
     <span dusk="value">{{ $value }}</span>
     <span dusk="events">{{ implode(',', $events) }}</span>
 
-    <button
```
Comment (ph7jack) [inline tests/Browser/Notifications/view.blade.php]: Please, roll back this file. I'm not seeing any changes here

File: ts/notifications/actions.ts [hunk 2]
```diff
@@ -45,6 +48,12 @@ export const parseActions = (options: ConfirmationOptions, componentId?: string)
       method: options.method,
       params: options.params
     }, options.accept)
+  } else if (options.emit) {
```
Comment (ph7jack) [inline ts/notifications/actions.ts]:
```suggestion
  }
  if (options.emit) {
```

**Review by ph7jack — CHANGES_REQUESTED** [review]
Nice Job, could you fix these two comments for me?

File: tests/Browser/Notifications/view.blade.php (hunk 1 above)
Comment (putera) [reply tests/Browser/Notifications/view.blade.php]: Noted.

**Review by ph7jack — APPROVED** [review]
Good job @putera

**Comment by ph7jack** [comment]
@putera I recommend you sign your commits. It will be good for all projects that you contribute
https://docs.github.com/pt/authentication/managing-commit-signature-verification/signing-commits

**Pull request merged**